modules/libcom/src/iocsh/initHooks.h

```diff
@@ -36,34 +40,63 @@ typedef enum {
     initHookAfterCaServerRunning,
-    initHookAfterIocRunning,        /* End of iocRun/iocInit commands */
+    initHookAfterIocRunning,        /** \brief End of iocRun/iocInit commands */
 
-    initHookAtIocPause,             /* Start of iocPause command */
+    initHookAtIocPause,             /** \brief Start of iocPause command */
     initHookAfterCaServerPaused,
     initHookAfterDatabasePaused,
-    initHookAfterIocPaused,         /* End of iocPause command */
+    initHookAfterIocPaused,         /** \brief End of iocPause command */
 
-    initHookAtShutdown,             /* Start of iocShutdown commands */
+    initHookAtShutdown,             /** \brief Start of iocShutdown commands */
     initHookAfterCloseLinks,
-    initHookAfterStopScan,          /* triggered only by unittest code.   testIocShutdownOk() */
-    initHookAfterStopCallback,      /* triggered only by unittest code.   testIocShutdownOk() */
+    initHookAfterStopScan,          /** \brief triggered only by unittest code.   testIocShutdownOk() */
+    initHookAfterStopCallback,      /** \brief triggered only by unittest code.   testIocShutdownOk() */
     initHookAfterStopLinks,
-    initHookBeforeFree,             /* triggered only by unittest code.   testIocShutdownOk() */
-    initHookAfterShutdown,          /* End of iocShutdown commands */
+    initHookBeforeFree,             /** \brief triggered only by unittest code.   testIocShutdownOk() */
+    initHookAfterShutdown,          /** \brief End of iocShutdown commands */
 
-/* Deprecated states, provided for backwards compatibility.
+/**
+ * Deprecated states, provided for backwards compatibility.
  * These states are announced at the same point they were before,
  * but will not be repeated if the IOC gets paused and restarted.
  */
-    initHookAfterInterruptAccept,   /* After initHookAfterDatabaseRunning */
-    initHookAtEnd,                  /* Before initHookAfterIocRunning */
+    initHookAfterInterruptAccept,   /** \brief After initHookAfterDatabaseRunning */
+    initHookAtEnd,                  /** \brief Before initHookAfterIocRunning */
 } initHookState;
 
+/** 
+ * Any functions that are registered before iocInit reaches the desired state will be called when it reaches that state. 
+ * The initHookName function returns a static string representation of the state passed into it which is intended for printing. The following skeleton code shows how to use this facility:
+ *\n
+ * static initHookFunction myHookFunction; \n
+ *\n
+ *int myHookInit(void) \n
+ * { \n
+ *  return(initHookRegister(myHookFunction)); \n
+ *} \n
+ * \n
+ * static void myHookFunction(initHookState state) \n
+ * { \n
+ *  switch(state) { \n 
+ *    case initHookAfterInitRecSup: \n
+ *      ... \n
+ *      break; \n
+ *   case initHookAfterInterruptAccept: \n
+ *      ... \n
+ *      break; \n
+ *    default: \n
+ *      break; \n
+ *  } \n
+ * } \n
+ * An arbitrary number of functions can be registered.
```

Comment:
If you want to input code in Doxygen comments, you put the code between `\code{.cpp}` and `\endcode` (no `\n` needed). Awesome example, in any case.